acquia/cli
# Mock application test network requests

**Motivation**
Application tests were making actual Cloud API requests, which made them somewhat flaky and impossible to mock or test for anything other than an unauthorized response. This also necessitated maintaining CA certs in Travis, which has been its own kind of trouble (https://github.com/acquia/cli/pull/408)

**Proposed changes**
Mock the request so that tests are faster and deterministic, and remove Travis CA certs so we don't have to maintain them. Use this pattern to extend test coverage with additional application tests, which is more robust than testing the ExceptionListener in isolation like we used to.

Mocking test services necessitated making all of our Symfony services public, which seems to be a known issue and best practice for CLI apps like ours that don't inherit from the Framework Bundle (and thus don't have access to the dedicated `test.service_container`).

**Testing steps**
How can we replicate the issue and verify that this PR fixes it?

1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `acli ckc`
3. Run `netstat` to see open connections.
4. `./vendor/bin/phpunit tests/phpunit/src --filter=ExceptionApplicationTest`
5. Run `netstat` to verify that there are no new connections.

**Merge requirements**
